### DISCUSSION TITLE
netbox-floorplan plugin permission issue

### DISCUSSION THREAD

**Discussion by ashokthangaraj** [netbox-community/netbox#16310] [Help Wanted!]
Hi All, I am running netbox-floorplan-plugin==0.3.6 and l am getting permission issue for all users except super user. 
My Netbox version is v3.7.0
Is it working for others?

Reference Slack discussion,
https://netdev-community.slack.com/archives/C01P0FRSXRV/p1716431990646149

**Comment by candlerb** [comment]
It would be helpful if you showed your actual error, and what you were doing that caused the error, rather than vague "permission issue".

A quick look on the issue tracker shows this: https://github.com/netbox-community/netbox-floorplan-plugin/issues/21
and the fix is here: https://github.com/netbox-community/netbox-floorplan-plugin/commit/b332f7edbe239d06c20cc7cf976bd2ff4b0996d5

This may or may not be your problem (it relates specifically to the permission for adding a floorplan)

**Comment by ashokthangaraj** [comment]
@candlerb I am getting Access Denied and "You do not have permission to access this page." error when i try to Add floorplan for the location.
<img width="1469" alt="Screenshot 2024-05-28 at 11 24 39 AM" src="https://github.com/netbox-community/netbox/assets/26494670/c67e5617-a276-4473-a241-28d3f58d9ae0">

I have a separate group and permission for floorplan with create,edit,update, delete permission without any constraints.

**Reply by candlerb** [reply]
> when i try to Add floorplan for the location.

That's the issue I linked to above then. The fix is a one-line edit to a python file, until a new version of the plugin is released.

**Reply by ashokthangaraj** [reply]
@candlerb can you please share me the link.I am not able to get from the above response.

**Reply by candlerb** [reply]
> A quick look on the issue tracker shows this: https://github.com/netbox-community/netbox-floorplan-plugin/issues/21
> and the fix is here: https://github.com/netbox-community/netbox-floorplan-plugin/commit/b332f7edbe239d06c20cc7cf976bd2ff4b0996d5

Those blue things are links. You click on them.

**Reply by ashokthangaraj** [reply]
@candlerb I seen this thread and as stated in the https://github.com/netbox-community/netbox-floorplan-plugin/issues/21#issuecomment-2116916034  "so pypi users has no chance to get a working version" I have installed using pypi and there is no file _netbox_floorplan/views.py_ i can find to edit the typo error.Please guide me.

**Reply by candlerb** [reply]
What did you do to try to find the file? I suggest something like this:

```
find /opt/netbox/venv -name views.py
```

Also, please stop pinging me personally. Any Linux system administrator should be able to help you.
